Unable to compile with OpenSSL on OpenBSD

### Describe the bug

During debug for issue #22239 on OpenBSD, I'm trying to compile `vlib/net/http/http_test.v` with OpenSSL. But compilation fails with unknown symbol `SSL_get1_peer_certificate`.

### Reproduction Steps

On OpenBSD current/amd64, try to run `vlib/net/http/http_test.v` with OpenSSL using `-d use_openssl`.

### Expected Behavior

No error for compilation with OpenSSL.

### Current Behavior

Compilation fails
```bash
$ ./v -d use_openssl vlib/net/http/http_test.v
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/http_test.01JVVMT9BCQPE4NA70AWC3YCMN.tmp.c:17197: warning: cast between pointer and integer of different size
cc: tcc: error: undefined symbol 'SSL_get1_peer_certificate'
...
cc: tcc: error: undefined symbol 'SSL_get1_peer_certificate'
(note: the original output was 4 lines long; it was truncated to its first 2 lines + the last line)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .

```

### Possible Solution

On OpenBSD, [LibreSSL](https://www.libressl.org/) (fork of OpenSSL after Heartbleed "apocalypse") is used by default. Some symbols/functions from OpenSSL are not available in LibreSSL (partial compatibility). **OpenSSL is only available in OpenBSD ports**. 

But in `vlib/net/openssl/openssl.c.v`, `cflags` and `libs` are not defined for OpenBSD case (finding OpenSSL from ports) => add code (via `pkgconfig`) to set them with OpenSSL from OpenBSD ports.

### Additional Information/Context

_No response_

### V version

V 0.4.10 e2cb01d

### Environment details (OS name and version, etc.)

|V full version      |V 0.4.10 e2cb01db880910bcec403b243be3df3101ab5aaf
|:-------------------|:-------------------
|OS                  |openbsd, 7.7, GENERIC.MP#663
|Processor           |4 cpus, 64bit, little endian
|Memory              |1.31GB/3.98GB
|                    |
|V executable        |/home/fox/dev/vlang.git/v
|V last modified time|2025-05-22 07:39:30
|                    |
|V home dir          |OK, value: /home/fox/dev/vlang.git
|VMODULES            |OK, value: /home/fox/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/fox/dev/vlang.git
|                    |
|Git version         |git version 2.49.0
|V git status        |weekly.2025.21-15-ge2cb01db
|.git/config present |true
|                    |
|cc version          |OpenBSD clang version 16.0.6
|gcc version         |egcc (GCC) 11.2.0
|clang version       |OpenBSD clang version 16.0.6
|tcc version         |tcc version 0.9.28rc 2024-09-14 HEAD@b8b6a5fd (x86_64 OpenBSD)
|tcc git status      |thirdparty-openbsd-amd64 8205cc59
|emcc version        |N/A
|glibc version       |N/A

--- a/vlib/net/openssl/openssl_openbsd.c.v
+++ b/vlib/net/openssl/openssl_openbsd.c.v
@@ -1,0 +1,5 @@
+module openssl
+
+// SSL_get_peer1_certificate not defined in LibreSSL (OpenSSL fork) on OpenBSD,
+// use SSL_get_peer_certificate instead.
+fn C.SSL_get_peer_certificate(ssl &SSL) &C.X509

--- a/vlib/net/openssl/ssl_connection.c.v
+++ b/vlib/net/openssl/ssl_connection.c.v
@@ -223,6 +223,7 @@ fn (mut s SSLConn) complete_connect() ! {
 	}
 
 	if s.config.validate {
+		mut pcert := &C.X509(unsafe { nil })
 		for {
 			mut res := C.SSL_do_handshake(voidptr(s.ssl))
 			if res == 1 {
@@ -239,7 +240,11 @@ fn (mut s SSLConn) complete_connect() ! {
 			}
 			return error('Could not validate SSL certificate. (${err_res}),err')
 		}
-		pcert := C.SSL_get1_peer_certificate(voidptr(s.ssl))
+		$if openbsd {
+			pcert = C.SSL_get_peer_certificate(voidptr(s.ssl))
+		} $else {
+			pcert = C.SSL_get1_peer_certificate(voidptr(s.ssl))
+		}
 		defer {
 			if pcert != 0 {
 				C.X509_free(pcert)

